### PR TITLE
Docker compose setup

### LINKED ISSUE #6: Add demo
- docker-compose file
- should run load balancer image
- should run polkadot node
- should run daemon connected to polkadot node and reporting data to load balancer

### LINKED ISSUE #11: Write instructions inside README file
Add instructions for:
- clone
- install dependencies
- run
- test

_Reformat README file_

### PR DESCRIPTION
**Pull request by MakMuftic**
### Changes

- Added `docker-compose` setup
- Refactored and expanded README file
- Refactored flags description 

### Issues

closes #6 
closes #11

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - "127.0.0.1:9615:9615/tcp"
+    image: chevdor/polkadot:latest
+    volumes:
+      - "polkadot-data:/data"
+    command: sh -c "polkadot --rpc-external --prometheus-external --log error --rpc-cors=all"
+
+  vedran:
+    build:
+      context: .
+      dockerfile: ./Dockerfile
```
Comment (mpetrunic) [inline docker-compose.yml]: Use published image. You can setup different compose for dev

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+version: '3'
+services:
+
+  polkadot:
+    ports:
+      - "127.0.0.1:30333:30333/tcp"
+      - "127.0.0.1:9933:9933/tcp"
+      - "127.0.0.1:9615:9615/tcp"
```
Comment (mpetrunic) [inline docker-compose.yml]: I don't think you need to expose ports

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  vedran:
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+    command: start --fee=0.1 --auth-secret=test-secret --log-level debug
+    image: nodefactory/vedran:${IMAGE_TAG:-latest}
+    container_name: "vedran"
+    ports:
+      - "${LB_LOCAL_PORT:-4000}:${LB_REMOTE_PORT:-4000}"
```
Comment (mpetrunic) [inline docker-compose.yml]: Remove this

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  vedran-daemon:
+    image: nodefactory/vedran-daemon:${IMAGE_TAG:-v0.1.1}
+    depends_on:
+      - vedran
+      - polkadot
+    entrypoint: /bin/sh -c "sleep 5; ./vedran-daemon --id test-id --lb http://vedran:4000 --node-rpc http://polkadot:9933 --node-metrics http://polkadot:9615 --payout-address 0xtestaddress --log-level error"
```
Comment (mpetrunic) [inline docker-compose.yml]: Why sleep? Demon should wait for node to start

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      context: .
+      dockerfile: ./Dockerfile
+    command: start --fee=0.1 --auth-secret=test-secret --log-level debug
+    image: nodefactory/vedran:${IMAGE_TAG:-latest}
+    container_name: "vedran"
+    ports:
+      - "${LB_LOCAL_PORT:-4000}:${LB_REMOTE_PORT:-4000}"
+
+  vedran-daemon:
+    image: nodefactory/vedran-daemon:${IMAGE_TAG:-v0.1.1}
```
Comment (mpetrunic) [inline docker-compose.yml]: Don't make it variable

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    volumes:
+      - "polkadot-data:/data"
+    command: sh -c "polkadot --rpc-external --prometheus-external --log error --rpc-cors=all"
+
+  vedran:
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+    command: start --fee=0.1 --auth-secret=test-secret --log-level debug
+    image: nodefactory/vedran:${IMAGE_TAG:-latest}
```
Comment (mpetrunic) [inline docker-compose.yml]: Set specific version without variable

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -19,24 +16,45 @@ Note that if you need to do this, you probably want to add your Go bin directory
...
+
+### Demo
+
+Dockerized demo with
+- _Polkadot node_
+- _Vedran daemon_
+- _Vedran loadbalancer_ 
 
-Expected name of the configuration file depends on `ENV` variable. For example, if you run a daemon app with `ENV=test go run main.go start`, expected config file name is `config-test.yaml`
+can be run with `docker-compose up`
```
Comment (mpetrunic) [inline README.md]: add note that docker and docker-compose are requirements along with links to installation

File: docker-compose.yml (hunk 4 above)
Comment (mpetrunic) [reply docker-compose.yml]: @MakMuftic Remove this and update version once @mpetrun5 fixes it

**Comment by MakMuftic** [comment]
@mpetrunic I will open separate _release PR_ after we close this one

**Comment by mpetrun5** [comment]
> @mpetrunic I will open separate _release PR_ after we close this one
@MakMuftic 
Do we need new release?
I think everything is already in 0,1.0.

**Comment by MakMuftic** [comment]
@mpetrun5 I changed log format for ping

**Review by mpetrun5 — DISMISSED** [review]

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    command: start --fee=0.1 --auth-secret=test-secret --log-level debug
+    image: nodefactory/vedran:v0.1.1
+    container_name: "vedran"
+
+  vedran-daemon:
+    image: nodefactory/vedran-daemon:v0.1.2
+    depends_on:
+      - vedran
+      - polkadot
+    entrypoint: /bin/sh -c "./vedran-daemon --id test-id --lb http://vedran:4000 --node-rpc http://polkadot:9933 --node-metrics http://polkadot:9615 --payout-address 0xtestaddress --log-level error"
```
Comment (mpetrunic) [inline docker-compose.yml]: should be:
command: --id test-id --lb http://vedran:4000 --node-rpc http://polkadot:9933 --node-metrics http://polkadot:9615 --payout-address 0xtestaddress --log-level error

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+version: '3'
+services:
+
+  polkadot:
+    image: chevdor/polkadot:latest
```
Comment (mpetrunic) [inline docker-compose.yml]: why aren't we using official image?
Pleaase use speicifc tag instead of latest

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 9 above)
Comment (MakMuftic) [reply docker-compose.yml]: I replaced it with official image

**Review by mpetrunic — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**